developer/src/kmc-convert/src/keylayout-to-kmn/keylayout-to-kmn-converter.ts

```diff
@@ -27,13 +27,46 @@ export interface convert_object {
   arrayOf_Rules: Rule[],
 };
 
-export class KeylayoutToKmnConverter {
+export interface allArrayContents_object {
+  actionId?: string,
+  keyCode?: string,
+  key?: string,
+  behaviour?: string,
+  modifier?: string,
+  outchar?: string;
+};
+
+export interface idStateOutput_object {
+  id: string,
+  state: string,
+  output: string,
+};
+
+/**
+ * @brief  member function to find the number of keys defined in a .keykayout file.
+ *         We process 'MAX_KEY_COUNT' keys at maximum. In case a keylayout has fewer keys defined, we use that smaller number of keys (USE_KEY_COUNT)
+ * @param  data data read from keylayout file
+ * @return static variable USE_KEY_COUNT holding the number of keys used in a .keykayout file.
+ */
+export function find_usedKeysCount(data: any, pos: number): number {
+  let USE_KEY_COUNT = KeylayoutToKmnConverter.MAX_KEY_COUNT;
+  for (let i = 0; i < data.keyboard.keyMapSet[0].keyMap.length; i++) {
+    //if (data.keyboard.keyMapSet[0].keyMap[i].key.length < USE_KEY_COUNT) {
+    if ((i === pos) && (data.keyboard.keyMapSet[0].keyMap[i].key.length < USE_KEY_COUNT)) {
```

Comment:
Why are you iterating through the array if you only do something at `pos`? Just eliminate the loop altogether?